rholang/src/test/scala/coop/rchain/rholang/interpreter/accounting/CostAccountingSpec.scala

```diff
@@ -52,11 +52,22 @@ class CostAccountingSpec extends FlatSpec with Matchers with PropertyChecks with
 
   val contracts = Table(
     ("contract", "expectedTotalCost"),
-    ("""@0!(2)""", 33L),
-    ("""@0!(2) | @1!(1)""", 69L),
-    ("""for(x <- @0){ Nil }""", 64L),
-    ("""for(x <- @0){ Nil } | @0!(2)""", 73L),
-    /*
+    ("""@0!(2)""", 97L),
+    ("""@0!(2) | @1!(1)""", 197L),
+    ("""for(x <- @0){ Nil }""", 128L),
+    ("""for(x <- @0){ Nil } | @0!(2)""", 329L),
+    ("@0!!(0) | for (_ <- @0) { 0 }", 342L),
+    ("@0!!(0) | for (x <- @0) { 0 }", 342L),
+    ("@0!!(0) | for (@0 <- @0) { 0 }", 336L),
```

Comment:
I added `("@0!!(0) | @1!!(1) | for (_ <- @0;_ <- @1) { 0 }", 586L)` and it looks to me like it's failing non-deterministically. Sometimes it expects 586, other times 594. Shouldn't this also be solved here?

```Cost was not repeatable, expected 594, got 586.```
```Cost was not repeatable, expected 586, got 594```

Reply:
How do you run your tests? sbt? Please paste the exact command so that there's less variables when I'm trying to reproduce that.

Reply:
As you've seen, the CI passed - and so did my local runs. Thx for catching those!

Reply:
Can you paste the full assertion output? Or even better a link to gist.github.com?

Reply:
Oh, that's a new test case! How did you arrive at it? Maybe you've got some intuitions what could be wrong that you used when creating it?

Reply:
I ran `sbt:rholang> testOnly *CostAccounting*`. Going to add a link to the gist in a moment

Reply:
Got it locally, thx. Any ideas how to test that better?

Reply:
https://gist.github.com/lukasz-golebiewski/151c65891b7ec3e8f9c14698e013cbe0

Reply:
@lukasz-golebiewski please have another look. Unless you can find any more bugs, I'd like to split additional testing (possibly generative) for cost repeatability into a separate PR.

Reply:
Sounds good!